pkg/pf/tfbridge/logging.go

```diff
@@ -29,6 +30,25 @@ import (
 func (p *provider) initLogging(ctx context.Context, sink logging.Sink, urn resource.URN) context.Context {
 	// add the resource URN to the context
 	ctx = tfbridge.XWithUrn(ctx, urn)
+
+	// There is no host in a testing context.
+	if sink == nil {
+		// For tests that did not call InitLogging yet, we should call it here so that
+		// GetLogger does not panic.
+		if ctx.Value(logging.CtxKey) == nil {
+			return logging.InitLogging(ctx, logging.LogOptions{
+				URN:             urn,
+				ProviderName:    p.info.Name,
+				ProviderVersion: p.info.Version,
+			})
+		}
+
+		// Otherwise keep the context as-is.
+		return ctx
+	}
+
+	log.SetOutput(tfbridge.NewTerraformLogRedirector(ctx, sink))
```

Comment:
This is almost copied verbatim from `pkg/tfbridge/provider.go`:

https://github.com/pulumi/pulumi-terraform-bridge/blob/893d7c88453da2248dcc866d70dd8de44f67a664/pkg/tfbridge/provider.go#L362-L379